server/static/dashboard/collab.js

```diff
@@ -7,42 +10,486 @@ collab.fetch(function(err) {
   document.querySelector('#partners').textContent = collab.data.users.slice().sort().join(' & ')
 });
 
+// Parameters required to use updateFunction
+// TODO: Reconcile parameters and extraArgs
+var parameters = {};
+
+// Whether deleted code is currently shown or not
+var showDeletedCode = false;
+
+
+//////////////////////////////////////
+///////////// MAIN CODE //////////////
+
+/**
+ * Fetch the files for this collab, determine which visual to use,
+ *   and display the files accordingly.
+ */
 connection.createFetchQuery('files', { collabid: collabid }, {}, function(err, files) {
   if (err) { throw err; }
-  
+
+  // Visual 1: total diff
+  // Visual 2: Regex matching
+  // Visual 3: total diff + regex matching
+  // Visual 4: total diff + hide common prefix
+  // Visual 5: total diff + regex matching + hide common prefix
+
+  if (visual[0] == '1' ||
+      visual[0] == '3' ||
+      visual[0] == '4' ||
+      visual[0] == '5') {
+    parameters["threshold"] = getThresholdFromUrl(visual);
+  }  
+
+  if (visual[0] == '2' ||
+      visual[0] == '3' ||
+      visual[0] == '5') {
+    var regexes = getRegexesFromUrl(visual);
+    parameters['regexes'] = regexes;
+    regexes.forEach(function(regex) {
+      addRegexToControls(regex);
+    });
+  }
+
+  if (visual[0] == '4' ||
+      visual[0] == '5') {
+    parameters['hideCommonPrefixes'] = true;
+  }
+
+  showFiles(files, parameters);
+});
+
+/**
+ * Display the files using the provided updateFunction and passing
+ *   extraArgs to that updateFunction.
+ */
+function showFiles(files, extraArgs) {
   var list = document.querySelector('#files');
   files.sort(function(a, b) { return a.data.filepath.localeCompare(b.data.filepath); });
   files.forEach(function(file) {
     var item = document.importNode(document.querySelector('#file').content, true);
+    
     var heading = item.querySelector('h4');
     heading.textContent = file.data.filepath;
     var diff = item.querySelector('.diff code');
     list.appendChild(item);
-    
+
     $.ajax('/baseline/' + project + '/' + file.data.filepath).done(function(baseline) {
+      // Save data for regex updating
+      // TODO: Better way to do this?
+      $(diff).data('baseline', baseline);
+      $(diff).data('text', file.data.text); // TODO: update this data on update
+      $(diff).data('filepath', file.data.filepath);
+
+      var extraArgsForFile = Object.assign({'filepath': file.data.filepath}, extraArgs);
+
       if (cutoff) {
         $.ajax('/historical/' + project + '/' + collabid + '/' + file.data.filepath + '/' + cutoff).done(function(historical) {
-          updateDiff(diff, baseline, historical.data ? historical.data.text : undefined);
+
+          // Used for regex updating
+          if (historical.data) {
+            $(diff).data('text', historical.data.text); // TODO: update this data on update
```

Comment:
Updates never happen, because the view with `cutoff` never updates?

Reply:
Removed.

Reply:
Update: Had to add this back in because even though updates never happen, we might change the regexes so we need to know what the filetext is.